django/contrib/syndication/views.py

```diff
@@ -83,9 +85,21 @@ def _get_dynamic_attr(self, attname, obj, default=None):
             # catching the TypeError, because something inside the function
             # may raise the TypeError. This technique is more accurate.
             try:
-                code = attr.__code__
+                func = unwrap(attr)
+                code = func.__code__
             except AttributeError:
-                code = attr.__call__.__code__
+                func = unwrap(attr.__call__)
+                code = func.__code__
+            # If function doesn't have arguments and it is not a staticmethod, it
+            # means was decorated without using `functools` (we need that to hold
+            # the reference to the original one).
+            if not code.co_argcount and not isinstance(
+                getattr_static(self, func.__name__, None), staticmethod
+            ):
+                raise ImproperlyConfigured(
+                    f"Feed method decorated by {func.__name__}() needs to use "
+                    "@functools.wraps"
+                )
```

Comment:
Maybe it would be better to catch `TypeError` instead of adding complicated checks :thinking: 
```python
try:
    if code.co_argcount == 2:  # one argument is 'self'
        return attr(obj)
    else:
        return attr()
except TypeError as e:
    raise ImproperlyConfigured(...) from e
```

Reply:
I didn't do it because of comment in line 84 to 86:

            # Check co_argcount rather than try/excepting the function and
            # catching the TypeError, because something inside the function
            # may raise the TypeError. This technique is more accurate.